### ISSUE TITLE
[k8s-deployment] Master process can fail on startup if FQDN is not resolvable yet

### ISSUE THREAD

**Issue by ajcaldera1** [YugaByte/yugabyte-db#677]
Noticed on GKE. Created one cluster of 4 n1-standard-16s using Yugaware to instantiate the universes.

First universe (test-1): 3 yb-master pods (2 cpu), 3 tserver pods (4 cpu) - 1 volume 100GB

Creation successful.

Second universe (test-2): 3 yb-master pods (2 cpu), 3 tserver pods (4 cpu) - 1 volume 100GB

Universe fails to create as one or more yb-masters fails to create with the following message:

```
I1220 13:33:51.777537     1 init.cc:110] version 1.1.9.0 build 1 revision d3c895e3384efd1bccc54ba626daf0496e42130d build_type RELEASE built at 19 Dec 2018 21:16:12 UTC
I1220 13:33:51.778090     1 mem_tracker.cc:240] MemTracker: hard memory limit is 3.398438 GB
I1220 13:33:51.778111     1 mem_tracker.cc:242] MemTracker: soft memory limit is 2.888672 GB
I1220 13:33:51.784497     1 master_main.cc:90] Initializing master server...
I1220 13:33:51.785696     1 fs_manager.cc:260] Opened local filesystem: /mnt/disk0
uuid: "5cb8cea19da54285928f0589d2f23fc2"
format_stamp: "Formatted at 2018-12-20 13:32:18 on yb-master-2"
I1220 13:33:51.786044     1 server_base.cc:172] Auto setting FLAGS_num_reactor_threads to 16
I1220 13:33:51.789402     1 master_main.cc:93] Starting Master server...
I1220 13:33:51.792153     1 webserver.cc:147] Starting webserver on 0.0.0.0:7000
I1220 13:33:51.792196     1 webserver.cc:152] Document root: /home/yugabyte/www
I1220 13:33:51.792511     1 webserver.cc:239] Webserver started. Bound to: http://0.0.0.0:7000/
I1220 13:33:51.792783     1 rpc_server.cc:167] RPC server started. Bound to: 10.4.1.8:7100
I1220 13:33:51.792840     1 server_base.cc:424] Using private ip address 10.4.1.8
I1220 13:33:51.793519    35 sys_catalog.cc:198] Trying to load previous SysCatalogTable data from disk
E1220 13:33:51.793944    35 master.cc:222] Master@10.4.1.8:7100: Unable to init master catalog manager: Not found (yb/util/env_posix.cc:1011): Unable to initialize catalog manager: Failed to initialize sys tables async: Could not load tablet metadata from /mnt/disk0/yb-data/master/tablet-meta/00000000000000000000000000000000: /mnt/disk0/yb-data/master/tablet-meta/00000000000000000000000000000000: No such file or directory (error 2)
E1220 13:33:51.794364     1 main_util.cc:22] Not found (yb/util/env_posix.cc:1011): Unable to initialize catalog manager: Failed to initialize sys tables async: Could not load tablet metadata from /mnt/disk0/yb-data/master/tablet-meta/00000000000000000000000000000000: /mnt/disk0/yb-data/master/tablet-meta/00000000000000000000000000000000: No such file or directory (error 2)
I1220 13:33:51.794431     1 master.cc:265] Master@10.4.1.8:7100 shutting down...
I1220 13:33:51.806542     1 master.cc:269] Master@10.4.1.8:7100 shutdown complete.
```

If you delete the aborting pod -- it recreates and throws the same error again. However, if you delete the second universe, and recreate it with the same parameters (including the same namespace/universe-name) that you tried (test-2) again, it succeeds.

**Comment by bbaddepudi** [comment]
`master_discovery_timeout_ms` flag could be reused for this purpose as well...

**Comment by bmatican** [comment]
Fixed by b55b44d

**Comment by kmuthukk** [comment]
Thx @bmatican & @bbaddepudi for find & quick resolution!
